test/all.spec.ts

```diff
@@ -318,6 +334,95 @@ describe('validation', () => {
 		};
 		expect(f).to.throw();
 	});
+
+	it('should not fail when build config specifies valid network', async () => {
+		const f = () => {
+			compose.normalize({
+				version: '2.4',
+				networks: {
+					mynet: {},
+				},
+				services: {
+					main: {
+						build: { context: '.', network: 'mynet' },
+					},
+				},
+			});
+		};
+		expect(f).to.not.throw();
+	});
+
+	it('should fail when build config specifies invalid network', async () => {
+		const f = () => {
+			compose.normalize({
+				version: '2.4',
+				networks: {
+					othernet: {},
+				},
+				services: {
+					main: {
+						build: { context: '.', network: 'mynet' },
+					},
+				},
+			});
+		};
+		expect(f).to.throw();
+	});
+
+	it('should support extension fields', async () => {
+		const f = () => {
+			const c = compose.normalize({
+				version: '2.4',
+				services: {
+					main: {
+						build: '.',
+						'x-extension': true,
+					},
+				},
+			});
+		};
+		expect(f).to.not.throw();
+	});
+
+	it('should throw when long syntax tmpfs mounts are specified', async () => {
+		const f = () => {
+			const c = compose.normalize({
+				version: '2.4',
+				services: {
+					main: {
+						build: '.',
+						volumes: [{ type: 'tmpfs', target: '/tmp2' }],
+					},
+				},
+			});
+		};
+		expect(f).to.throw();
+	});
+
+	it(`should throw when long syntax volume mounts specify options`, async () => {
+		const f = () => {
+			const c = compose.normalize({
+				version: '2.4',
+				services: {
+					main: {
+						build: '.',
+						volumes: [
+							{
+								type: 'volume',
+								source: 'v1',
+								target: '/v1',
+								volume: { nocopy: true },
+							},
+						],
```

Comment:
Same question as just above.

Reply:
currently:

`Mount options are not allowed`

same as above, should the correct way to do this be in the error?

Reply:
I don't know off the top of my head if the two cases are identical. In this case (I think) we literally don't support a docker-compose feature, it's not a case of "we do support it but in a different way", right? If so, then the error looks good to me as it is -- there's nowhere to point the user to.

Reply:
yeah, this is about the options under the `volumes[].volume` field, so actually only `nocopy` at v2.4. I will also call this 'volume options' instead to align with the docker docs